Formula/opencv.rb

```diff
@@ -92,43 +116,41 @@ def install
       -DWITH_VTK=ON
       -DBUILD_opencv_python2=OFF
       -DBUILD_opencv_python3=ON
-      -DPYTHON3_EXECUTABLE=#{Formula["python@3.9"].opt_bin}/python3
+      -DPYTHON3_EXECUTABLE=#{which(python3)}
     ]
 
     # Disable precompiled headers and force opencv to use brewed libraries on Linux
     if OS.linux?
-      args << "-DENABLE_PRECOMPILED_HEADERS=OFF"
-      args << "-DJPEG_LIBRARY=#{Formula["libjpeg"].opt_lib}/libjpeg.so"
-      args << "-DOpenBLAS_LIB=#{Formula["openblas"].opt_lib}/libopenblas.so"
-      args << "-DOPENEXR_ILMIMF_LIBRARY=#{Formula["openexr"].opt_lib}/libIlmImf.so"
-      args << "-DOPENEXR_ILMTHREAD_LIBRARY=#{Formula["openexr"].opt_lib}/libIlmThread.so"
-      args << "-DPNG_LIBRARY=#{Formula["libpng"].opt_lib}/libpng.so"
-      args << "-DPROTOBUF_LIBRARY=#{Formula["protobuf"].opt_lib}/libprotobuf.so"
-      args << "-DTIFF_LIBRARY=#{Formula["libtiff"].opt_lib}/libtiff.so"
-      args << "-DWITH_V4L=OFF"
-      args << "-DZLIB_LIBRARY=#{Formula["zlib"].opt_lib}/libz.so"
+      args += %W[
+        -DENABLE_PRECOMPILED_HEADERS=OFF
+        -DJPEG_LIBRARY=#{Formula["jpeg-turbo"].opt_lib}/libjpeg.so
+        -DOpenBLAS_LIB=#{Formula["openblas"].opt_lib}/libopenblas.so
+        -DOPENEXR_ILMIMF_LIBRARY=#{Formula["openexr"].opt_lib}/libIlmImf.so
+        -DOPENEXR_ILMTHREAD_LIBRARY=#{Formula["openexr"].opt_lib}/libIlmThread.so
+        -DPNG_LIBRARY=#{Formula["libpng"].opt_lib}/libpng.so
+        -DPROTOBUF_LIBRARY=#{Formula["protobuf"].opt_lib}/libprotobuf.so
+        -DTIFF_LIBRARY=#{Formula["libtiff"].opt_lib}/libtiff.so
+        -DWITH_V4L=OFF
+        -DZLIB_LIBRARY=#{Formula["zlib"].opt_lib}/libz.so
+      ]
     end
 
-    if Hardware::CPU.intel?
-      args << "-DENABLE_AVX=OFF" << "-DENABLE_AVX2=OFF"
-      args << "-DENABLE_SSE41=OFF" << "-DENABLE_SSE42=OFF" unless MacOS.version.requires_sse42?
+    # Ref: https://github.com/opencv/opencv/wiki/CPU-optimizations-build-options
+    ENV.runtime_cpu_detection
+    if Hardware::CPU.intel? && build.bottle?
+      cpu_baseline = MacOS.version.requires_sse42? ? "SSE4_2" : "SSSE3"
+      args += %W[-DCPU_BASELINE=#{cpu_baseline} -DCPU_BASELINE_REQUIRE=#{cpu_baseline}]
     end
```

Comment:
I think for non-bottle builds, it should:
- on Linux, `-march=native` should force CPU detection
   https://github.com/opencv/opencv/blob/4.6.0/cmake/OpenCVCompilerOptimizations.cmake#L162-L166 
- on macOS, the default is auto-detection
   https://github.com/opencv/opencv/blob/4.6.0/cmake/OpenCVCompilerOptimizations.cmake#L319-L321

Reply:
This might need adjusting for source builds, then, since `ENV.runtime_cpu_detection` will skip passing any `-march` flags.

Reply:
In that case, Linux install will probably be similar between bottles and source build as SSE3 is default baseline which should give same result as SSSE3.

There is a `CPU_BASELINE=NATIVE` which I didn't see any documentation on but may provide a more tuned variant.

Reply:
I'm ok to set `CPU_BASELINE=NATIVE` if we're not building bottles as long as it doesn't lead to dependent weirdness as with the Facebook formulae.